src/mscorlib/shared/System/BitConverter.cs

```diff
@@ -92,8 +105,15 @@ public static byte[] GetBytes(long value)
             return bytes;
         }
 
-        public static bool TryWriteBytes(Span<byte> destination, long value) =>
-            WriteUnaligned(value, destination);
+        // Cpmverts a long into a Span
+        public static bool TryWriteBytes(Span<byte> destination, long value)
+        {
+            if (destination.Length < Unsafe.SizeOf<long>())
+                return false;
+
+            destination[0] = (byte)value;
```

Comment:
So the fix instead of line 114 would be: 
`Unsafe.WriteUnaligned(ref destination.DangerousGetPinnableReference(), value);` ?